Add grpc_status_number as a default access log field

rajatvig opened the pull request:
Fixes https://github.com/projectcontour/contour/issues/4484

Signed-off-by: Rajat Vig <rvig@etsy.com>

rajatvig commented:
Could this be reviewed please?

skriss commented on changelogs/unreleased/4880-rajatvig-small.md:
```diff
@@ -0,0 +1 @@
+Add `grpc_status_code` to the log message field
```
```suggestion
Add `grpc_status_number` to the default JSON access log fields
```

skriss reviewed (changes requested):
One minor comment on the changelog, otherwise LGTM.

cc @sunjayBhatia

sunjayBhatia commented on apis/projectcontour/v1alpha1/accesslog.go:
```diff
@@ -43,6 +43,7 @@ var DefaultAccessLogJSONFields = AccessLogJSONFields([]string{
 	"user_agent",
 	"x_forwarded_for",
 	"grpc_status",
+	"grpc_status_number",
```
huh seems like we already added this to `envoySimpleOperators` in this file

sunjayBhatia replied:
(the access log format operator version)

sunjayBhatia reviewed (approved):
might want to be more judicious about the default access log fields in the future, but seems good to add this one

sunjayBhatia commented:
> might want to be more judicious about the default access log fields in the future, but seems good to add this one

lgtm minus @skriss' comments I should say

rajatvig commented:
> > might want to be more judicious about the default access log fields in the future, but seems good to add this one
> 
> lgtm minus @skriss' comments I should say

Addressed the comments. Agree, the default access log fields should only be judiciously added.

skriss commented:
Thanks @rajatvig!
